### ISSUE TITLE
IntentTEDPolicy: Implement loss function

### ISSUE THREAD

**Issue by dakshvar22** [RasaHQ/rasa#8438]
**Description of Problem**:
`IntentTEDPolicy` is designed to predict multiple labels instead of just one. Hence, the current `DotProductLoss` layer is unusable as it assumes only a single label to be correct for each data point.

**Overview of the Solution**:
Implement a new loss layer inherited from `DotProductLoss` which takes the above into account. The new loss function should use sigmoid cross entropy as opposed to softmax cross entropy (to accomodate multiple label learning/prediction).

A draft version of this is already implemented in `itedp-for-merge` branch.

[Scope Document](https://docs.google.com/document/d/1kjZD59NtnvAkxSfu6sjLzgDucTQT8LJhdfZnILCDVcE/edit?usp=sharing)

[Implementation proposal](https://www.notion.so/rasa/Implementation-Proposal-IntentTedPolicy-f16c3eb785964b1ca1249d508f6dc234#6ceb8d8b71024e27b94c1809dcdc22d7)

**Definition of Done**:
- [ ] Layer implemented as described
- [ ] Tests are added
